zigpy/ota/provider.py

```diff
@@ -161,6 +162,101 @@ class Trådfri(Basic):
         return key.manufacturer_id != self.MANUFACTURER_ID
 
 
+@attr.s
+class LedvanceImage:
+    """Ledvance image handler."""
+
+    OTA_HEADER = 0x0BEEF11E .to_bytes(4, "little")
+
+    manufacturer_id = attr.ib()
+    image_type = attr.ib()
+    version = attr.ib(default=None)
+    image_size = attr.ib(default=None)
+    url = attr.ib(default=None)
+
+    @classmethod
+    def new(cls, data):
+        ident = data["identity"]
+        ver = ident["version"]
+
+        res = cls(ident["company"], ident["product"])
+        res.file_version = int(data["fullName"].split("/")[1], 16)
+        res.image_size = data["length"]
+
+        res.url = "https://api.update.ledvance.com/v1/zigbee/firmwares/download"
+        res.url += "?Company=" + str(ident["company"]) + "&Product="
+        res.url += str(ident["product"]) + "&Version=" + str(ver["major"]) + "."
+        res.url += str(ver["minor"]) + "." + str(ver["build"])
+
+        return res
+
+    @property
+    def key(self):
+        return ImageKey(self.manufacturer_id, self.image_type)
+
+    async def fetch_image(self) -> Optional[OTAImage]:
+        async with aiohttp.ClientSession() as req:
+            LOGGER.debug("Downloading %s for %s", self.url, self.key)
+            async with req.get(self.url) as rsp:
+                data = await rsp.read()
+
+        assert len(data) > 24
+        offset = data.index(self.OTA_HEADER)
+        assert offset >= 0
+
+        img, _ = OTAImage.deserialize(data[offset:])
```

Comment:
Are Ledvance OTA images not starting with the OTA header? the Ikea images were not standard, wrapped in another container and needed the checks above. If they are standard OTA images you can skip lines 203--205 and just do `img, _ = OTAImage.deserialize(data)`

Reply:
The Ledvance images do start with the header. Like you inferred, I followed the IKEA provider. Will remove the check

Reply:
Don't worry about it for now.